### PR TITLE
fix invalid assert during property enumeration

### PR DESCRIPTION
**Pull request by MikeHolman**
In case cache was disabled, PropertyValueInfo might not have the expected PropertyString. Change assert to ignore this case.

OS: 12327564

### CODE REVIEW

File: lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp [hunk 1]
```diff
@@ -253,7 +253,7 @@ namespace Js
 
         if (info.GetPropertyString() != nullptr && info.GetPropertyString()->ShouldUseCache())
         {
-            Assert(propertyString == info.GetPropertyString());
+            Assert(info.IsNoCache() || propertyString == info.GetPropertyString());
```
Comment (rajatd) [inline lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp]: >IsNoCache [](start = 24, length = 9)

if the cache was disabled, is it ok to cache the property string on it?

Comment (MikeHolman) [inline lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp]: I thought it was fine, and would be a nop, but probably better off to add this check as a condition of the if anyway.

Comment (rajatd) [inline lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp]: Ok, just that the new condition is duplicated :)

---
In reply to: [126219071](https://github.com/Microsoft/ChakraCore/pull/3153#discussion_r126219071) [](ancestors = 126219071)

Comment (MikeHolman) [inline lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp]: oops, fixed!

**Comment by rajatd** [comment]
:shipit:

**Pull request merged**